draft-ietf-tls-rfc4492bis.xml

```diff
@@ -101,7 +101,8 @@
         generation of bulk encryption/MAC keys and initialization vectors is independent of the key exchange
         algorithm and not impacted by the introduction of ECC.</t>
       <t> <xref target="tbl2"/> summarizes the new key exchange algorithms.  All of these key exchange algorithms
-        provide forward secrecy.</t>
+        provide forward secrecy if and only if fresh ephemeral keys are generated and used, and also destroyed
```

Comment:
We got a good description in Seoul (and previously in emails) how you can use ECDHE and not get FS. Simply use the same server key-share all the time, and save it to a file so that you can later decrypt captured traffic.  So just choosing a ciphersuite that has _ECDHE_ in its name does not guarantee FS. You also need to generate and destroy keys.

I intentionally did not say that you have to generate the fresh keys for each and every handshake (my code generates a new one every ten seconds). Besides, you get the FS not from generating the fresh keys, but from destroying the used keys.